src/servers/src/http/event.rs

```diff
@@ -271,18 +263,47 @@ pub async fn log_ingester(
     .await
 }
 
+fn extract_pipeline_value_by_content_type(
+    content_type: ContentType,
+    payload: String,
+    ignore_errors: bool,
+) -> Result<PipelineValue> {
+    Ok(match content_type {
+        ct if ct == ContentType::json() => {
+            let json_value = transform_ndjson_array_factory(
+                Deserializer::from_str(&payload).into_iter(),
+                ignore_errors,
+            )?;
+
+            PipelineValue::try_from(json_value)
+                .map_err(|reason| CastTypeSnafu { msg: reason }.build())
+                .context(PipelineSnafu)?
+        }
+        ct if ct == ContentType::text() || ct == ContentType::text_utf8() => {
+            let arr = payload
+                .lines()
+                .filter(|line| !line.is_empty())
+                .map(|line| {
+                    let mut map = HashMap::new();
+                    map.insert("line".to_string(), PipelineValue::String(line.to_string()));
+                    PipelineValue::Map(map.into())
```

Comment:
Can we use some like `PipelineValue::TextLines(line)` here? Adding this map can have same overhead as `json!`

Reply:
Just noticed we have `PipelineValue::String`. But I think we don't need to use internal value type here. Perhaps we can create another enum say `PipelineDataSource` which has variants like `Json`, `Text`, `CSV` and even `DLT` in future.

Reply:
pipeline does not currently support parsing plain text. This is because pipelines currently need to support keys to select what to parse.

Reply:
I think what you suggest here is a input datasource trait for pipeline engine, that allows custom value extraction using field name in pipeline processing. Since we can have multiple processors along the line, the datatype before and after data is processed should be same and consistent, so that processors can be arranged in arbitrary order. 

Simply add an enum for indicating input datatype at entry level doesn't seem to benefit a lot since we still need to convert data into some compatible format during processor line. Maybe this is a topic for refactoring the whole pipeline datatype system that we can discuss later.